javatests/arcs/showcase/references/Arcs.kt

```diff
@@ -35,116 +76,77 @@ class Arcs(
         override fun getLifecycle() = lifecycle
     }
 
-
-    val arcHost = ArcHost(
+    val activationFactory = ServiceStoreFactory(
         context,
         fakeLifecycleOwner.lifecycle,
-        schedulerProvider,
-        connectionFactory
+        connectionFactory = connectionFactory
     )
 
-    private lateinit var allocator: Allocator
+    val arcHost = ArcHost(
+        schedulerProvider,
+        activationFactory = activationFactory
+    )
 
-    /**
-     * Start the arc containing the particles that provide the persistence. This should be run
-     * one time per instance, via [startIfNot].
-     */
-    private suspend fun startArc() {
-        /** This should probably live in an initializer somewhere. */
-        val hostRegistry = ExplicitHostRegistry().apply {
+    val hostRegistry = ExplicitHostRegistry().apply {
+        runBlocking {
             registerHost(arcHost)
         }
-
-        allocator = Allocator.create(
-            hostRegistry,
-            EntityHandleManager(
-                arcId = "allocator",
-                hostId = "allocator",
-                time = JvmTime,
-                scheduler = schedulerProvider.invoke("allocator")
-            )
-        )
-
-        allocator.startArcForPlan(WriteRecipePlan)
     }
 
-    suspend fun stop() {
-        allocator.stopArc("!:testArc".toArcId())
-    }
-
-    private var started = false
-    val mutex = Mutex()
-
-    private suspend fun startIfNot() = mutex.withLock {
-        if (!started) {
-            startArc()
-            started = true
-        }
-    }
+    val allocator = Allocator.create(
+        hostRegistry,
+        EntityHandleManager(
+            arcId = "allocator",
+            hostId = "allocator",
+            time = JvmTime,
+            scheduler = schedulerProvider.invoke("allocator")
+        )
+    )
 
-    fun all0(): List<MyLevel0> {
-        return runBlocking {
-            startIfNot()
-            arcHost.reader0.read()
-        }
+    suspend fun <T : Particle> getParticle(
+        plan: Plan,
+        particleName: String
+    ): T {
+        val arc = startArc(plan)
+        return getParticle(arc.id, particleName)
     }
 
-    fun put0(item: MyLevel0) {
-        runBlocking {
-            startIfNot()
-            arcHost.writer0.write(item)
-        }
+    suspend fun startArc(plan: Plan): Arc {
+        val arc = allocator.startArcForPlan(plan)
+        arc.waitForStart()
+        return arc
     }
 
-    fun all1(): List<MyLevel1> {
-        return runBlocking {
-            startIfNot()
-            arcHost.reader1.read()
-        }
+    suspend fun stopArcForPlan(plan: Plan) {
+        // Start the arc so that we have an instance of an arc to call stop on. Note this means that
+        // calling stop on an unstarted Arc will result in the Arc being started and immediately
+        // stopped.
+        val allocatorArc = allocator.startArcForPlan(plan)
+        allocatorArc.stop()
```

Comment:
What happens if we call stop but it's not fully started? See comments above.

I also wonder whether this logic is at risk due to arc lifecycle race condition risk per https://github.com/PolymerLabs/arcs/pull/5575

Reply:
The best logic here would be to look up an existing Arc for a given ID, and then stop that if it exists. I could not find a way to get an `Arc` by ID.

Reply:
(There should be a waitForStart here)

Reply:
Hmmm, on further investigation, it seems that `AbstractArcHost.stopArc` is synchronous, so I think I can just convert this all to a `allocator.stopArc(plan.arcId)` call.

Reply:
(reminded me that we want to file a similar task for storage proxy listeners, as well!)